pal/audio/minigame/player-minigame.ts

```diff
@@ -225,14 +247,16 @@ export class AudioPlayerMinigame implements OperationQueueable {
     @enqueueOperation
     seek (time: number): Promise<void> {
         return new Promise((resolve) => {
-            time = clamp(time, 0, this.duration);
-            if (time === 0 && this.currentTime === 0) {
-                // skip invalid seek
-                resolve();
-            } else {
-                this._eventTarget.once(AudioEvent.SEEKED, resolve);
+            // KNOWN ISSUES: on Baidu: currentTime returns without numbers on decimal places
+            if (this._state === AudioState.PLAYING && !this._seeking) {
+                time = clamp(time, 0, this.duration);
+                this._seeking = true;
                 this._innerAudioContext.seek(time);
+            } else if (this._cacheTime !== time) { // Skip the invalid seek
+                this._cacheTime = time;
+                this._needSeek = true;
             }
+            resolve();
```

Comment:
if we called innerAudioContext.seek(), please resolve in the callback of onSeek()

Reply:
Actually, if we use cache time to save middle state, we can directly resolve here.